packages/core/src/components/popover/popover.tsx

```diff
@@ -555,39 +557,39 @@ export class Popover extends AbstractComponent<IPopoverProps, IPopoverState> {
             this.handleMouseLeave(e);
         } else if (!this.props.isDisabled) {
             // only begin opening popover when it is enabled
-            this.setOpenState(true, e, this.props.hoverOpenDelay);
+            this.setOpenStateByPropCallback(true, e, this.props.hoverOpenDelay);
         }
     };
 
     private handleMouseLeave = (e: React.SyntheticEvent<HTMLElement>) => {
         // user-configurable closing delay is helpful when moving mouse from target to popover
-        this.setOpenState(false, e, this.props.hoverCloseDelay);
+        this.setOpenStateByPropCallback(false, e, this.props.hoverCloseDelay);
     };
 
     private handlePopoverClick = (e: React.MouseEvent<HTMLElement>) => {
         const eventTarget = e.target as HTMLElement;
         const shouldDismiss = eventTarget.closest(`.${Classes.POPOVER_DISMISS}`) != null;
         const overrideDismiss = eventTarget.closest(`.${Classes.POPOVER_DISMISS_OVERRIDE}`) != null;
         if (shouldDismiss && !overrideDismiss) {
-            this.setOpenState(false, e);
+            this.setOpenStateByPropCallback(false, e);
         }
     };
 
     private handleOverlayClose = (e: React.SyntheticEvent<HTMLElement>) => {
         const eventTarget = e.target as HTMLElement;
         // if click was in target, target event listener will handle things, so don't close
         if (!Utils.elementIsOrContains(this.targetElement, eventTarget) || e.nativeEvent instanceof KeyboardEvent) {
-            this.setOpenState(false, e);
+            this.setOpenStateByPropCallback(false, e);
         }
     };
 
     private handleTargetClick = (e: React.MouseEvent<HTMLElement>) => {
         // ensure click did not originate from within inline popover before closing
         if (!this.props.isDisabled && !this.isElementInPopover(e.target as HTMLElement)) {
             if (this.props.isOpen == null) {
-                this.setState(prevState => ({ isOpen: !prevState.isOpen }));
+                this.setOpenStateDirectly(previousIsOpen => !previousIsOpen);
```

Comment:
Can we do:

```tsx
this.setOpenStateDirectly(!this.state.previousIsOpen);
```

and then get rid of the helper's function-type argument altogether?